### ISSUE TITLE
Admin search results: show more info

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#10645]
Steps: login as admin and search for a student.

Current:
![image](https://user-images.githubusercontent.com/1673303/91067777-eb2e8000-e665-11ea-9bd1-dcf32125c93c.png)

Suggestions:
* Link student name to the course  of that student in the course instructor (i.e., masquerade as the instructor of the course) 
  * Reason: This link was used to quickly reach the session/course details when students send us support requests
* ~Give the full links i.e., starting with `https://teammates...`~
  * ~Reason: I copy-paste these links to preview the submission page, when students request us support requests~
  * Done

**Comment by damithc** [comment]
> Give the full links i.e., starting with `https://teammates...`
> * Reason: I copy-paste these links to preview the submission page, when students request us support requests

This is creating extra work for me, especially when re-sending links to students. After copy pasting the link to the email, I have to add the base URL by hand (multiplied by a few hundred times a month) I assume it's easy to fix, so let's do that soon. If both can't be done quickly, do this one first and leave the other problem to another PR.

**Comment by wkurniawan07** [comment]
> multiplied by a few hundred times a month

The fact that you had to do this a few hundred times a month tells another problem?

**Comment by damithc** [comment]
> The fact that you had to do this a few hundred times a month tells another problem?

Hopefully, this number will go down as now we have the link recovery feature. But sometimes I have to manually resend the link using a different email account because our regular email account may be blocked by some spam filter used by the user and they don't know how to unblock it.

**Comment by wkurniawan07** [comment]
Reverting priority to low as the more important part has been done.

**Comment by fans2619** [comment]
@wkurniawan07 
@damithc As suggested by the previous PR, I have a rough idea of how to bring you to the course page as an instructor of that course. This is doable because it seems the current system restricts there to be at least one instructor for any course. However, the system doesn't seem to have the restriction that there is at least one instructor having the permission to view student details. This means you potentially won't be able to view the student information (as that instructor) even if you're brought to the page as an instructor.

**Comment by samuelfangjw** [comment]
> This is doable because it seems the current system restricts there to be at least one instructor for any course

I don't think this is true as #9534 exists.

**Comment by fans2619** [comment]
Yeah probably that's a even worse case.

**Comment by wkurniawan07** [comment]
> However, the system doesn't seem to have the restriction that there is at least one instructor having the permission to view student details.

This is one of the reasons why I take out the "good first issue" label.

Courses with no instructor is possible in theory, but in practice not a cause of concern because only admin can leave a course in such state.

**Comment by fans2619** [comment]
I'm going to take this issue.

**Comment by fans2619** [comment]
![image](https://user-images.githubusercontent.com/68739475/169634749-4dd35536-06d5-4a25-ab5b-bcd0e56fb500.png)

@damithc By saying "link to the course instructor", is there a specific instructor you want to link to, since a course may have more than one instructors? Or do you have other preferred ways instead of a direct link?

**Comment by damithc** [comment]
> @damithc By saying "link to the course instructor", is there a specific instructor you want to link to, since a course may have more than one instructor? Or do you have other preferred ways instead of one direct link?

Good question. Perhaps prick one the course owners at random?

**Comment by fans2619** [comment]
That sounds a fair enough solution but I'm just concerned there may not be any co-owner at all. Of course in reality I think there should always be at least one co-owner for any course but that's not what the system restricts now. So perhaps failing that checking (with a small probability), it's a bit hard to decide which instructor to link to then.

**Comment by damithc** [comment]
> That sounds a fair enough solution but I'm just concerned there may not be any co-owner at all. Of course in reality I think there should always be at least one co-owner for any course but that's not what the system restricts now. So perhaps failing that checking (with a small probability), it's a bit hard to decide which instructor to link to then.

Sure, failing that, you can pick any of instructors at random.

**Comment by fans2619** [comment]
Sure just you may not have the permission you actually need in that case. 🤧

**Comment by damithc** [comment]
> Sure just you may not have the permission you actually need in that case. 🤧

It's fine. This is a convenience feature anyway (there are other ways to achieve the same). But if not too complicated, you can prioritize based on permissions.

**Comment by fans2619** [comment]
Okay!
